deployment/environment/memory/zksync_chains.go

```diff
@@ -0,0 +1,85 @@
+package memory
+
+import (
+	"context"
+	"math/big"
+	"strconv"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/hashicorp/consul/sdk/freeport"
+	"github.com/stretchr/testify/require"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/zksync-sdk/zksync2-go/accounts"
+	"github.com/zksync-sdk/zksync2-go/clients"
+
+	chain_selectors "github.com/smartcontractkit/chain-selectors"
+
+	"github.com/smartcontractkit/chainlink-testing-framework/framework/components/blockchain"
+	"github.com/smartcontractkit/chainlink/deployment"
+)
+
+func GenerateChainsZk(t *testing.T, numChains int) map[uint64]deployment.Chain {
+	chains := make(map[uint64]deployment.Chain)
+
+	for i := 0; i < numChains; i++ {
+		chainID := chain_selectors.TEST_90000051.EvmChainID + uint64(i) //nolint:gosec // it shouldn't overflow
```

Comment:
This is highly unlikely but still safe to add a comment in newmemory chains that TEST_90000051 onwards is reserved for zkchains if you have both chains in the env

Reply:
Added this comment 37efa67ce6e15dfedf2702cb66b4615d4d04a4c2